problem_builder/models.py

```diff
@@ -35,11 +35,19 @@ class Answer(models.Model):
     """
 
     class Meta:
-        unique_together = (('student_id', 'course_id', 'name'),)
+        unique_together = (
+            ('student_id', 'course_id', 'name'),
+            ('student_id', 'course_key', 'name'),
+        )
 
     name = models.CharField(max_length=50, db_index=True)
     student_id = models.CharField(max_length=32, db_index=True)
-    course_id = models.CharField(max_length=255, db_index=True)
+    # course_id is deprecated; it will be removed in next release.
+    course_id = models.CharField(max_length=50, db_index=True)
+    # course_key is the new course_id replacement with extended max_length.
+    # We need to allow NULL values during the transition period,
+    # but will be set to null=False in next release.
+    course_key = models.CharField(max_length=255, db_index=True, null=True)
```

Comment:
No, multiple `NULL` values are not considered to be identical to each other so they don't violate the uniqueness constraint.